File: library/swift/stats/Distribution.swift
import Foundation

/// A time series distribution tracking quantile/sum/average stats.
@objc
public protocol Distribution: AnyObject {
  /// Record a new value to add to the integer distribution.
  func recordValue(value: Int)

  /// Gets a distribution with tags attached.
  func attach(tags: [Tag]) -> Distribution
}

File: library/kotlin/io/envoyproxy/envoymobile/stats/Distribution.kt
package io.envoyproxy.envoymobile

/** A time-series distribution that tracks quantile/sum/average stats. */
interface Distribution : Stats {

  /** Records a new value to add to the distribution. */
  fun recordValue(value: Int)

  /** Gets a distribution with tags attached. */
  fun attach(tags: List<Tag>): Distribution
}
